tailcfg/tailcfg_view.go

```diff
@@ -1,13 +1,120 @@
+// Copyright (c) 2022 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Code generated by tailscale.com/cmd/cloner; DO NOT EDIT.
+//go:generate go run tailscale.com/cmd/cloner -type=User,Node,Hostinfo,NetInfo,Login,DNSConfig,RegisterResponse,DERPRegion,DERPMap,DERPNode -clonefunc
+
 package tailcfg
 
 import (
 	"encoding/json"
 	"errors"
+	"time"
 
+	"inet.af/netaddr"
+	"tailscale.com/types/dnstype"
+	"tailscale.com/types/key"
 	"tailscale.com/types/opt"
 	"tailscale.com/types/views"
 )
 
+// View returns a readonly view of User.
+func (p *User) View() UserView { return UserView{p} }
+
+type UserView struct {
```

Comment:
generate docs for the types at least? and mention that methods are only safe to call if Valid?